app/uk/gov/hmrc/iossreturns/services/ReturnsService.scala

```diff
@@ -102,24 +110,25 @@ class ReturnsService @Inject()(clock: Clock, vatReturnConnector: VatReturnConnec
       getAllPeriodsUntil(startPeriod :: periodsUpToNow, startPeriod.getNext(), endPeriod)
   }
 
-  def decideStatus(period: Period, vatReturn: Option[EtmpVatReturn], exclusions: List[EtmpExclusion]) = {
+  def decideStatus(period: Period, fulfilledPeriods: List[Period], exclusions: List[EtmpExclusion]): PeriodWithStatus = {
     if (isPeriodExcluded(period, exclusions))
       PeriodWithStatus(period, SubmissionStatus.Excluded)
-    else
-      vatReturn match {
-        case Some(_) => PeriodWithStatus(period, SubmissionStatus.Complete)
-        case None => if (LocalDate.now(clock).isAfter(period.paymentDeadline)) {
+    else {
+      if (fulfilledPeriods.contains(period)) {
+        PeriodWithStatus(period, SubmissionStatus.Complete)
+      } else {
+        if (LocalDate.now(clock).isAfter(period.paymentDeadline)) {
           PeriodWithStatus(period, SubmissionStatus.Overdue)
         } else {
           PeriodWithStatus(period, SubmissionStatus.Due)
         }
       }
+    }
   }
 
   def getLastExclusionWithoutReversal(exclusions: List[EtmpExclusion]): Option[EtmpExclusion] = {
-
-      exclusions.maxByOption(_.effectiveDate).filterNot(_.exclusionReason == Reversal)
-
+    // Even though API is array ETMP only return single item
+    exclusions.headOption.filterNot(_.exclusionReason == Reversal)
```

Comment:
in other places lastoption is used

https://github.com/hmrc/ioss-returns-frontend/blob/368f31ed19017f23a8ee7650bcec5b8e6eccbd0c/app/controllers/YourAccountController.scala#L51

Should probably agree on last of first, else weirdness?

Reply:
doesn't matter which, there will only ever be one